tools/template_based_query_generation/src/main/java/QueryType.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Types of queries in SQL language encoding
+ */
+public enum QueryType {
+    QUERY_ROOT,
+    DML,
+    DQL,
+    DDL,
+    SINK,
+    DDL_CREATE,
+    DDL_PARTITION,
+    DDL_CLUSTER,
```

Comment:
DML/DQL/DDL does not seem to be in the equivalent type as DDL_CREATE/DDL_PARTITION/... Consider splitting?

Reply:
I see what you mean: two types are needed but the tricky part is MarkovChain and Node need objects of the same type. What I plan is to have to still keep MarkovChain<Query> and Node<Query> but I will change the structure of Query in a later PR (adding stuff about number of tokens needed, etc)

Reply:
Please add a todo here.

Reply:
Yup added!